### PR TITLE
rename build-tests to release-test and make it required

### PR DESCRIPTION
**Pull request by rkpagadala**
make the release test required so that daily builds are not broken

### CODE REVIEW

**Comment by rkpagadala** [comment]
/hold
until https://github.com/istio/istio/pull/9893 is merged

File: prow/cluster/jobs/istio/istio/istio.istio.master.yaml [hunk 1]
```diff
@@ -218,11 +218,10 @@ presubmits:
           - entrypoint
           - prow/e2e-simpleTests-cni.sh
         nodeSelector:
-    - name: build-tests
+    - name: release-test
```
Comment (hklai) [inline prow/cluster/jobs/istio/istio/istio.istio.master.yaml:221]: move this to postsubmit for now.

failure like the one we had recently is just noise to be in presubmit. also you should set up testgrid to see historical results.

Comment (rkpagadala) [reply prow/cluster/jobs/istio/istio/istio.istio.master.yaml:221]: Our daily builds failed for both 1.1 and master, having this in presubmit avoids that pain.
Will setup testgrid for this job.

**Review by hklai — APPROVED** [review]

**Review by hklai — APPROVED** [review]
/lgtm

**Pull request merged**